sdk/src/Services/EC2/Custom/Model/IpPermission.cs

```diff
@@ -74,7 +74,15 @@ internal List<IpRange> PreIpv4Ranges
         /// <returns></returns>
         internal IpRangeValue CanModify()
         {
-            List<string> ipv4Ranges = this._ipv4Ranges.Select(p => p.CidrIp).ToList();
+#pragma warning disable CS0618
+            if (this._ipv4Ranges != null && this.IpRanges == null)
```

Comment:
I don't follow this change... Are we returning a different value now? I tried reading the `summary` but it seems out-of-date (for example, it mentions `ArgumentException` which this method doesn't throw at all).

Reply:
I updated the doc comment to reflect the return is actually `IpRangeValue.Invalid` in the case of an invalid situation. 

The changes I made were to add the `pragma` to get rid of the warnings for calling obsolete methods and checks if one or both collection is null at the start. If only one has a value and the other is null then we know to use the one with a value. If both are null default `Ipv4Ranges` which is the same behavior as if both collections are empty.

This is some confusing code dealing with the obsolete `IpRanges` property and might be worth removing in V4.